tests/test_free_energy.py

```diff
@@ -289,6 +291,28 @@ def test_plot_pair_bar_plots(mock_fig, hif2a_ligand_pair_single_topology_lam0_st
     assert set(mock_fig.call_args.args[0]) == expected_potentials
 
 
+@pytest.mark.nocuda
+@pytest.mark.parametrize("num_windows", [2, 5])
+def test_get_water_sampler_params(num_windows):
+    mol_a, mol_b, core = get_hif2a_ligand_pair_single_topology()
+    # Use the simple charges simply because it is faster
+    forcefield = Forcefield.load_from_file("smirnoff_1_1_0_sc.py")
+    st = SingleTopology(mol_a, mol_b, core, forcefield)
+    solvent_sys, solvent_conf, solvent_box, solvent_top = builders.build_water_system(3.0, forcefield.water_ff)
+    num_host_atoms = solvent_conf.shape[0]
+    host_system, masses = convert_omm_system(solvent_sys)
+    solvent_host = Host(host_system, masses, solvent_conf, solvent_box, num_host_atoms)
+    for lamb in np.linspace(0.0, 1.0, num_windows, endpoint=True):
+        state = setup_initial_state(st, lamb, solvent_host, DEFAULT_TEMP, 2024)
+        water_sampler_nb_params = get_water_sampler_params(state)
+        nb_pot = next(p.potential for p in state.potentials if isinstance(p.potential, Nonbonded))
+        ligand_water_params = st._get_guest_params(
+            forcefield.q_handle_solv, forcefield.lj_handle_solv, lamb, nb_pot.cutoff
+        )
+
+        np.testing.assert_array_equal(water_sampler_nb_params[num_host_atoms:], ligand_water_params)
```

Comment:
can we sanity check that:

1) at lambda=0, dummy atoms of mol_b have w-coords set to cutoff?
2) at lambda=1, dummy_atoms of mol_a have w-coords set to cutoff?

you can use the look up the dummy_states from the single_topology.c_flags:

```
# 0: Core
# 1: R_A
# 2: R_B
```

Reply:
Added in https://github.com/proteneer/timemachine/pull/1286/commits/c9ddadeb7d99f1546cc70c93737c18367609ebe4